### ISSUE TITLE
For binned models, ds.predict() fails when newdata is given: binned set to TRUE in meta.data but distbegin and distend fields are missing

### ISSUE THREAD

**Issue by jjrob** [DistanceDevelopment/mrds#18]
Here's the repro:

```R
> library(mrds)
Error in library(mrds) : there is no package called ‘mrds’
> library(devtools)
> install_github("DistanceDevelopment/mrds")
Downloading GitHub repo DistanceDevelopment/mrds@master
from URL https://api.github.com/repos/DistanceDevelopment/mrds/zipball/master
Installing mrds
"C:/PROGRA~1/R/R-33~1.3/bin/i386/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  "C:/Users/jjr8/AppData/Local/Temp/3/RtmpSKM8NG/devtools128c705d7d3a/DistanceDevelopment-mrds-6c9a0cc"  \
  --library="C:/Users/jjr8/Documents/R/win-library/3.3" --install-tests 

* installing *source* package 'mrds' ...
** R
** data
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
*** arch - i386
*** arch - x64
* DONE (mrds)
> library(mrds)
This is mrds 2.1.18
Built: R 3.3.3; ; 2017-05-12 13:17:59 UTC; windows
> data(book.tee.data)
> egdata <- book.tee.data$book.tee.dataframe
> max(egdata$distance)
[1] 3.84
> egdata$distbegin <- floor(egdata$distance*2)/2
> egdata$distend <- egdata$distbegin + 0.5
> head(egdata, 10)
   object observer detected distance size sex exposure distbegin distend
1       1        1        1     2.68    2   1        1       2.5     3.0
21      1        2        0     2.68    2   1        1       2.5     3.0
2       2        1        1     3.33    2   1        0       3.0     3.5
22      2        2        0     3.33    2   1        0       3.0     3.5
3       3        1        1     0.34    1   0        0       0.0     0.5
23      3        2        0     0.34    1   0        0       0.0     0.5
4       4        1        1     2.53    2   1        1       2.5     3.0
24      4        2        0     2.53    2   1        1       2.5     3.0
5       5        1        1     1.46    2   1        0       1.0     1.5
25      5        2        0     1.46    2   1        0       1.0     1.5
> breaks = seq(0,4,0.5)
> breaks
[1] 0.0 0.5 1.0 1.5 2.0 2.5 3.0 3.5 4.0
> m <- ddf(dsmodel=~cds(key="hn", formula=~1), data=egdata, method="ds", meta.data=list(breaks=breaks, binned=TRUE, width=4))
> p <- predict(m)      # Works fine
> p <- predict(m, newdata=egdata)
Error in process.data(newdata, object$meta.data, check = FALSE) : 
  binned set to TRUE in meta.data but distbegin and distend fields are missing
```

I suspect the problem is that predict.ds is not including the distbegin and distend columns in the data frame it constructs around line 112 of predict.ds.R. It appears to be including just the formula terms and the distance column:

```R
          # pull out the columns in the formula and the distances column
          fvars <- all.vars(as.formula(model$ds$aux$ddfobj[[df_par]]$formula))

          if(!all(fvars %in% colnames(newdata))){
            stop("columns in `newdata` do not match those in fitted model\n")
          }

          model_dat <- model_dat[,c("distance", fvars), drop=FALSE]

          # setup the covariate matrix, using the model data to ensure that
          # the levels are right
          newdata <- rbind(model_dat,
                           newdata_save[,c("distance", fvars), drop=FALSE])
```

**Comment by dill** [comment]
Thanks for reporting this Jason. Can you try out aa85ee3 and see if that 
fixes this for you?

**Comment by CottonRockwood** [comment]
I ran into this issue in 2.1.17 and banged my head against it for a while... then I found Jason's bug report... Thanks, Jason!
It looks like this is fixed in 2.1.18, so should this be closed?

**Comment by jjrob** [comment]
Yes, sorry! I agree it should be closed. I patched my local copy but then never checked the git repo. 2.1.18 does appear to fix it. I'll close this now...
